src/drive/web/modules/actionmenu/ActionsItems.jsx

```diff
@@ -7,20 +7,28 @@ import styles from 'drive/styles/actionmenu.styl'
 export const ActionsItems = ({ actions, file, onClose }) => {
   const { t } = useI18n()
   let previousAction = ''
-  return actions.map((actionObject, i) => {
-    const actionName = Object.keys(actionObject)[0]
+  const displayableActions = actions.filter(actionObject => {
     const actionDefinition = Object.values(actionObject)[0]
-    if (
+    return !(
       actionDefinition.displayCondition !== undefined &&
       !actionDefinition.displayCondition([file])
     )
-      return null
+  })
+
+  return displayableActions.map((actionObject, i) => {
+    const actionName = Object.keys(actionObject)[0]
+    const actionDefinition = Object.values(actionObject)[0]
+
     if (previousAction === actionName) {
       previousAction = actionName
       return null
     } else {
       previousAction = actionName
     }
+
+    if (actionName === 'hr' && i + 1 === displayableActions.length) {
+      return null
+    }
```

Comment:
Why is there an hr at the end ? Can't we remove it from the data from the start ?

Reply:
We can't, since we don't know if the `hr` will be the latest action since we have condition to display an action. 

I added 15d19068d to clean the actions before this map and added a comment to explain why